amplify/backend/function/fetchMotionState/src/utils.js

```diff
@@ -170,12 +175,49 @@ const didMotionPass = ({
   return false;
 };
 
+const updateColonyUnclaimedStakes = async (
+  colonyAddress,
+  transactionHash,
+  motionId,
+  updatedStakerRewards,
+) => {
+  const { data } = await graphqlRequest(getColony, {
+    id: colonyAddress,
+  });
+
+  const motionsWithUnclaimedStakes =
+    data?.getColony?.motionsWithUnclaimedStakes ?? [];
+
+  const motionWithUnclaimedStake = motionsWithUnclaimedStakes?.find(
+    ({ transactionHash: txHash }) => txHash === transactionHash,
+  );
+
+  const unclaimedRewards = updatedStakerRewards.filter(
+    ({ isClaimed }) => !isClaimed,
+  );
+
+  if (!motionWithUnclaimedStake && unclaimedRewards.length) {
```

Comment:
If we have an unfinalizable motion with stakes (e.g a motion that fails before going to a vote), we check to see if there's already an entry in the `motionsWithUnclaimedStakes` array. 

If not, we check to see if there are any unclaimed stakes, and if so, we add the motion to the array. 

These checks as described are necessary because we remove the motion from the array in the block ingestor once it no longer has any unclaimed rewards. These checks stop us from re-adding it with an empty unclaimed rewards array.